benchexec/localexecution.py

```diff
@@ -260,7 +265,8 @@ def execute_benchmark(benchmark, output_handler):
             "System has swapped during benchmarking. "
             "Benchmark results are unreliable!"
         )
-
+    if pqos.reset_required and pqos.cli_exists:
```

Comment:
I would move the check for `reset_required` into the `Pqos` class (better encapsulation).

Reply:
I have set the field `reset_required` in the Pqos class itself. Are you saying to move the above two functions in `reset_resources` function itself?

Reply:
At last the check for `reset_required`, yes, because this seems like internal-only knowledge of that class. But `cli_exists` could be moved as well.